python/tank_vendor/shotgun_authentication/session_cache.py

```diff
@@ -411,9 +413,39 @@ def generate_session_token(hostname, login, password, http_proxy, auth_token=Non
         raise AuthenticationError("Authentication failed.")
     except (ProtocolError, httplib2.ServerNotFoundError):
         raise AuthenticationError("Server %s was not found." % hostname)
+    # In the following handlers, we are not rethrowing an AuthenticationError for
+    # a very specific reason. While wrong credentials or host is a user
+    # recoverable error, problems with proxy settings or network errors are much
+    # more severe errors which can't be fixed by reprompting. Therefore, they have
+    # nothing to do with authentication and shouldn't be reported as such.
+    except socket.error, e:
+        logger.exception("Unexpected connection error.")
+        # e.message is always an empty string, so look at the exception's arguments.
+        # The arguments are always a string or a number and a string.
+        if isinstance(e.args[0], str):
+            # if the error is just a string, simply forward the message.
+            raise Exception(e.args[0])
+        else:
+            # We could argue here that we should only display the string portion of the
+            # error since the error code is of little relevance to the user, but since
+            # Toolkit doesn't properly log everything to a file at the moment, it's probably
+            # safer to have the error code a little bit more in the open. Also, the formatting
+            # of this exception type is pretty bad so let's reformat it ourselves. By default, it
+            # turns a tuple into a string.
+            raise Exception("%s (%d)" % (e.args[1], e.args[0]))
+    except httplib2.socks.ProxyError, e:
+        logger.exception("Unexpected connection error.")
+        # Same comment applies here around formatting.
+        # Note that e.message is always a tuple in this
+        raise Exception("%s (%d)" % (e.message[1], e.message[0]))
     except MissingTwoFactorAuthenticationFault:
         # Silently catch and rethrow to avoid logging.
         raise
-    except:
+    except Exception, e:
         logger.exception("There was a problem logging in.")
-        raise
+        # If the error message is empty, like httplib.HTTPException, convert
+        # the class name to a string
+        if len(str(e)) == 0:
+            raise Exception("Unknown error: %s" % e.__class__.__name__)
```

Comment:
very minor but you should probably do type(e).__name__ here instead of e.__class__.__name__